src/Sulu/Component/Content/Types/BlockContentType.php

```diff
@@ -369,39 +381,26 @@ private function prepareData(PropertyInterface $property, callable $dataCallback
             $blockProperty = $blockProperty->getProperty();
         }
 
-        $data = [];
+        $blockPropertyTypes = [];
         for ($i = 0; $i < $blockProperty->getLength(); ++$i) {
             $blockPropertyType = $blockProperty->getProperties($i);
-            $blockPropertyTypeSettings = $blockPropertyType->getSettings();
 
-            if (
-                \is_array($blockPropertyTypeSettings)
-                && !empty($blockPropertyTypeSettings['hidden'])
-            ) {
-                continue;
-            }
+            foreach ($this->blockVisitors as $blockVisitor) {
```

Comment:
i would really like to extract this into a seperate PR that targets `release/2.2`

Reply:
If we need that I would still do that after this PR has been merged.